src/Files.Uwp/Views/MainPage.xaml.cs

```diff
@@ -383,6 +383,21 @@ private void Page_Loaded(object sender, RoutedEventArgs e)
             }
         }
 
+        private void Page_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            switch (Pane?.Position)
+            {
+                case PanePositions.Right when ContentColumn.ActualWidth == ContentColumn.MinWidth:
+                    UserSettingsService.PaneSettingsService.VerticalSizePx += e.NewSize.Width - e.PreviousSize.Width;
```

Comment:
When you change the value of a setting, the whole cache is serialized. This will cause massive performance issues as well as storage strain